sentry-spring-boot-starter/src/main/java/io/sentry/spring/boot/SentryProperties.java

```diff
@@ -0,0 +1,325 @@
+package io.sentry.spring.boot;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.core.SentryLevel;
+import io.sentry.core.SentryOptions;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * Configuration for Sentry in a shape of Spring Boot configuration bean. Example:
+ *
+ * <pre>
+ * sentry:
+ *     enabled: true
+ *     dsn: https://00059966e6224d03a77ea5eca10fbe18@sentry.mycompany.com/14
+ *     release: "1.0.1"
+ *     dist: x86
+ *     environment: staging
+ *     server-name: megaServer
+ * </pre>
+ */
+@ConfigurationProperties("sentry")
+@Open
+public class SentryProperties {
```

Comment:
Isn't this the same properties class used on logback? is this something we can push to sentry-core or something to avoid redeclaring it?

Reply:
I would prefer to keep it as a refactoring outside of this PR.

Reply:
yeah it makes sense to me, otherwise, when we deprecate or add something, it'd need to be done on different classes and easy to forget.

Reply:
We have two options:

1. Create `SentryProperties` in `core` module that will contain subset of properties from `SentryOptions` and a method `applyTo(SentryOptions)` which will copy values from `SentryProperties` to `SentryOptions`.

2. Extract subset of properties from `SentryOptions` into new class - `SentryProperties` and add field `properties` in `SentryOptions` - this means that everywhere in the code where we are currently referring to one of these properties we will need to additionally do `getProperties().getXXX`, for example: `options.getProperties().isDebug()`.

Option 1 - is relatively simple to add and would affect only backend-related integrations but does not solve the duplication between `SentryProperties` and `SentryOptions`.
Option 2- requires changing all the code where `SentryOptions` is called but completely removes any duplication of `SentryOptions` properties.

Which direction is preferred? @bruno-garcia @marandaneto

Reply:
Option 1 works for me. It can also land as a separate refactor PR as you said.

One other proposal would be to have SentryOptions inherit from SentryProperties. We can use (or create if not there) the overload that takes the instance as opposed to the callback for `init`. 

The code that binds values to SentryProperties would then take the instance of SentryOptions instead, and there wouldn't be the need of a `applyTo` or any code duplication.

Would that work?

Reply:
I'd say that @bruno-garcia suggestion looks good in theory, let's experiment it if we have a method like `applyTo` we'd still need to take care of it in every change of protocol anyway.
so ideally no code duplication nor breaking changes for Android unless we don't find another way.

Reply:
the downside of extracting this to code though is that you can't use the class `Optional` like you did and looks nice.
this could be a separate PR, so we have this working this way for now.

Reply:
Agreed. So lets merge then